db/src/query.ts

```diff
@@ -0,0 +1,444 @@
+    return this.field(k) as any;
+  }
+
+  /**
+   * Equals
+   * @param x - value for comparison
+   * @return - a filter
+   */
+  public eq(x: Equatable): EqFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'eq',
+      path: this.parts,
+      value: x,
+    };
+  }
+
+  /**
+   * Not equals
+   * @param x - value for comparison
+   * @return - a filter
+   */
+  public ne(x: Equatable): NeFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'ne',
+      path: this.parts,
+      value: x,
+    };
+  }
+
+  /**
+   * Greater than
+   * @param x - value for comparison
+   * @return - a filter
+   */
+  public gt(x: Comparable): GtFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'gt',
+      path: this.parts,
+      value: x,
+    };
+  }
+
+  /**
+   * Greater than or equals
+   * @param x - value for comparison
+   * @return - a filter
+   */
+  public gte(x: Comparable): GteFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'gte',
+      path: this.parts,
+      value: x,
+    };
+  }
+
+  /**
+   * Less than
+   * @param x - value for comparison
+   * @return - a filter
+   */
+  public lt(x: Comparable): LtFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'lt',
+      path: this.parts,
+      value: x,
+    };
+  }
+
+  /**
+   * Less than or equals
+   * @param x - value for comparison
+   * @return - a filter
+   */
+  public lte(x: Comparable): LteFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'lte',
+      path: this.parts,
+      value: x,
+    };
+  }
+
+  /**
+   * Does path exist?
+   * @return - a filter
+   */
+  public exists(): ExistsFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'exists',
+      path: this.parts,
+    };
+  }
+
+  /**
+   * Is value at path null?
+   * @return - a filter
+   */
+  public isNull(): IsNullFilter {
+    return {
+      [filterSymbol]: true,
+      operator: 'isNull',
+      path: this.parts,
+    };
+  }
+
+  /**
+   * String matches pattern.
+   * @param pattern regular expression
+   * @param caseInsensitive should the check be case insensitive?
+   * @return - a filter
+   */
+  public matches(pattern: string, caseInsensitive?: boolean): MatchesFilter;
+
+  /**
+   * String matches pattern.
+   * @param pattern - regular expression, if the RegExp object has the 'i' flag, perform a case insensitive match.
+   * @return - a filter
+   */
+  public matches(pattern: RegExp): MatchesFilter;
+
+  public matches(pattern: RegExp | string, caseInsensitive: boolean = false): MatchesFilter {
+    if (typeof pattern === 'string') {
+      return {
+        [filterSymbol]: true,
+        operator: 'matches',
+        path: this.parts,
+        pattern,
+        caseInsensitive,
+      };
+    } else if (pattern instanceof RegExp) {
+      if (/[^i]/.test(pattern.flags)) {
+        throw new TypeError('Given RegExp flags not supported');
```

Comment:
```suggestion
        throw new TypeError('Only /i RegExp flag supported');
```
I prefer to state what is allowed, for about the same character count.  Because even an error message can be positive.